static/app/views/settings/organizationIntegrations/integrationButton.tsx

```diff
@@ -0,0 +1,83 @@
+import {useContext} from 'react';
+
+import {Button} from 'sentry/components/button';
+import {IconOpen} from 'sentry/icons';
+import type {Integration} from 'sentry/types/integrations';
+import type {Organization} from 'sentry/types/organization';
+import {AddIntegrationButton} from 'sentry/views/settings/organizationIntegrations/addIntegrationButton';
+import {IntegrationContext} from 'sentry/views/settings/organizationIntegrations/integrationContext';
+import RequestIntegrationButton from 'sentry/views/settings/organizationIntegrations/integrationRequest/RequestIntegrationButton';
+
+type Props = {
+  buttonProps: ButtonProps;
+  onAddIntegration: (integration: Integration) => void;
+  onExternalClick: () => void;
+  organization: Organization;
+  userHasAccess: boolean;
+  externalInstallText?: string;
+};
+
+type ButtonProps = {
+  disabled?;
+  priority?;
+  size?;
+  style?;
+} | null;
+
+function IntegrationButton({
+  organization,
+  userHasAccess,
+  onAddIntegration,
+  onExternalClick,
+  externalInstallText,
+  buttonProps,
+}: Props) {
+  const integration = useContext(IntegrationContext);
+  if (!integration) {
+    return null;
+  }
+  const {provider, type, installStatus, analyticsParams, modalParams} = integration;
+  const {metadata} = provider;
```

Comment:
nit: can safely destruct `integration` like this:
```suggestion
  const {provider, type, installStatus, analyticsParams, modalParams } = useContext(IntegrationContext) ?? {};
  if (!provider) return null;
  const {metadata} = provider;
```